test/controllers/DisableLoadingPartialsSpec.scala

```diff
@@ -0,0 +1,171 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.CFConfig
+import connectors.deskpro.HmrcDeskproConnector
+import connectors.enrolments.EnrolmentsConnector
+import org.mockito.Matchers.any
+import org.mockito.Mockito.when
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.mockito.MockitoSugar.mock
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.http.Status.NOT_FOUND
+import play.api.i18n.MessagesApi
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.mvc.{MessagesControllerComponents, Result}
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{defaultAwaitTimeout, status}
+import uk.gov.hmrc.play.bootstrap.tools.Stubs
+import util.{BackUrlValidator, RefererHeaderRetriever}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DisableLoadingPartialsSpec extends AnyWordSpec with GuiceOneAppPerSuite with Matchers {
+
+  override def fakeApplication(): Application =
+    new GuiceApplicationBuilder()
+      .configure(
+        "metrics.jvm"      -> false,
+        "metrics.enabled"  -> false,
+        "auditing.enabled" -> false,
+        "disablePartials"  -> true
+      )
+      .build()
+
+  "ReportProblemControllerSpec" should {
+    "returns 404 when partialIndex is called " in new TestScope {
+      val result: Future[Result] = reportProblemController.partialIndex(
+        preferredCsrfToken = None,
+        service = Some(serviceName)
+      )(FakeRequest())
+
+      status(result) should be(NOT_FOUND)
+    }
+
+    "returns 404 when partialAjaxIndex is called" in new TestScope {
+      val result: Future[Result] = reportProblemController.partialAjaxIndex(
+        service = Some(serviceName)
+      )(FakeRequest())
+
+      status(result) should be(NOT_FOUND)
+    }
+  }
+
+  "ContactHmrcController" should {
+    "returns 404 when partialIndex is called" in new TestScope {
+      val result: Future[Result] = contactHmrcController.partialIndex(
+        submitUrl = "",
+        csrfToken = "",
+        service = Some(serviceName),
+        renderFormOnly = true
+      )(FakeRequest())
+
+      status(result) should be(NOT_FOUND)
+    }
+  }
+
+  "FeedbackController" should {
+    "returns 404 when partialIndex is called" in new TestScope {
+      val result: Future[Result] = feedbackController.partialIndex(
+        submitUrl = "",
+        csrfToken = "",
+        service = Some(serviceName),
+        referer = None,
+        canOmitComments = true,
+        referrerUrl = None
+      )(FakeRequest())
+
+      status(result) should be(NOT_FOUND)
+    }
+  }
+
+  trait TestScope {
```

Comment:
there's a lot of boilerplate for these tests, since we are starting the application with GuiceOneAppPerSuite is it possible to provide the mocks to the application then just get the automatically wired up controller?
```
app.injector.instanceOf[ReportProblemController]
```